File: 2019/python/day05.py
#%%
from intcode import IntCode
from utils import load

raw = load("day05.txt", split=",", parseint=True)

#%% Part 1
def run(x):
    ic = IntCode(raw, x)
    ic.execute()
    return ic.outputs[-1]


def test05a(benchmark):
    assert benchmark(run, [1]) == 15426686


def test05b(benchmark):
    assert benchmark(run, [5]) == 11430197

File: 2019/rust/src/day05.rs
use super::intcode::*;

fn run_ic(raw: &[String], code: isize) -> IntCode {
    let mut ic = IntCode::from(&raw[0]);
    ic.input.push_front(code);
    ic.run();
    ic
}

pub fn part1(raw: &[String]) -> usize { run_ic(raw, 1).output.pop_back().unwrap() as usize }

pub fn part2(raw: &[String]) -> usize { run_ic(raw, 5).output.pop_back().unwrap() as usize }

#[cfg(test)]
mod tests {
    use super::*;
    use crate::utils::*;
    #[test]
    fn test1() {
        assert_eq!(part1(&read("day05.txt")), 15426686);
    }

    #[test]
    fn test2() {
        assert_eq!(part2(&read("day05.txt")), 11430197);
    }
}
